src/app_service/service/token/service.nim

```diff
@@ -290,80 +293,78 @@ QtObject:
-      var tokenSymbols: seq[string] = @[]
+      var allTokens: Table[string, TokenItem] = initTable[string, TokenItem]()
+      var tokensByName: Table[string, TokenBySymbolItem] = initTable[string, TokenBySymbolItem]()
+      var uniqueNonCustomTokenSymbols: HashSet[string] = initHashSet[string]()
 
+      self.sourcesOfTokensList = @[]
       for s in tokenList.data:
+        var
+          uniqueTokensPerList: HashSet[string] = initHashSet[string]()
+          uniqueNamesPerList: HashSet[string] = initHashSet[string]()
+        # token type determined by the source
+        let tokenType = if s.name == NativeTokensListName: TokenType.Native else: TokenType.ERC20
+        for token in s.tokens:
+          if not supportedNetworkChains.contains(token.chainID):
+            continue
+
+          let tokenKey = token.tokenKey()
+          uniqueTokensPerList.incl(tokenKey)
+          uniqueNamesPerList.incl(token.byNameModelKey())
+          if allTokens.hasKey(tokenKey):
+            allTokens[tokenKey].sources.add(s.name)
+          else:
+            allTokens[tokenKey] = TokenItem(
+              key: tokenKey,
+              name: token.name,
+              symbol: token.symbol,
+              sources: @[s.name],
+              chainID: token.chainID,
+              address: token.address,
+              decimals: token.decimals,
+              image: token.image,
+              `type`: tokenType,
+              communityId: token.communityData.id
+            )
+
+          let tokenByNameKey = token.byNameModelKey()
+          if tokensByName.hasKey(tokenByNameKey):
+            if not tokensByName[tokenByNameKey].sources.contains(s.name):
+              tokensByName[tokenByNameKey].sources.add(s.name)
+            let tokenChainId = token.chainID
+            if tokensByName[tokenByNameKey].addressPerChainId.filter(apc => apc.chainId == tokenChainId).len == 0:
+              tokensByName[tokenByNameKey].addressPerChainId.add(AddressPerChain(chainId: token.chainID, address: token.address))
+          else:
+            tokensByName[tokenByNameKey] = TokenBySymbolItem(
+              key: tokenByNameKey,
+              name: token.name,
+              symbol: token.symbol,
+              sources: @[s.name],
+              addressPerChainId: @[AddressPerChain(chainId: token.chainID, address: token.address)],
+              decimals: token.decimals,
+              image: token.image,
+              `type`: tokenType,
+              communityId: token.communityData.id
+            )
+
+        if s.name != CustomTokensListName:
+          uniqueNonCustomTokenSymbols = uniqueNonCustomTokenSymbols + uniqueTokensPerList
+
         let newSource = SupportedSourcesItem(
           name: s.name,
           updatedAt: s.lastUpdateTimestamp,
           source: s.source,
           version: s.version,
-          tokensCount: s.tokens.len
+          tokensCount: uniqueNamesPerList.len
```

Comment:
Since we grouped tokens across chains the number decreased a bit.

<img width="1740" alt="image" src="https://github.com/user-attachments/assets/670daa96-4ef3-4928-85a9-ea34a9af9a57" />